File: leetcode/299-bulls-and-cows.cpp
class Solution
{
 public:
  string getHint(string secret, string guess)
  {
    int bull = 0, cow = 0;
    vector<int> cnt(10, 0);
    for (int i = 0; i < secret.size(); ++i) {
      int c0 = secret[i] - '0', c1 = guess[i] - '0';
      if (c0 == c1) ++bull;
      else {
        cow += (cnt[c0] < 0) + (cnt[c1] > 0);
        ++cnt[c0], --cnt[c1];
      }
    }
    return to_string(bull) + "A" + to_string(cow) + "B";
  }
};

File: leetcode/299-bulls-and-cows.py
class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """
        bull, cow = 0, 0
        cnt = {str(c): 0 for c in range(10)}
        for c0, c1 in zip(secret, guess):
            if c0 == c1:
                bull += 1
            else:
                if cnt[c0] < 0:
                    cow += 1
                if cnt[c1] > 0:
                    cow += 1
                cnt[c0] += 1
                cnt[c1] -= 1
        return '{0}A{1}B'.format(bull, cow)
